bundles/org.openhab.binding.sensibo/src/main/java/org/openhab/binding/sensibo/internal/handler/SensiboSkyHandler.java

```diff
@@ -0,0 +1,448 @@
+    public static final String FAN_LEVEL_PROPERTY = "fanLevel";
+    public static final String MODE_PROPERTY = "mode";
+    public static final String TARGET_TEMPERATURE_PROPERTY = "targetTemperature";
+    public static final String SWING_MODE_LABEL = "Swing Mode";
+    public static final String FAN_LEVEL_LABEL = "Fan Level";
+    public static final String MODE_LABEL = "Mode";
+    public static final String TARGET_TEMPERATURE_LABEL = "Target Temperature";
+    private final Logger logger = LoggerFactory.getLogger(SensiboSkyHandler.class);
+    private final Map<ChannelTypeUID, ChannelType> generatedChannelTypes = new HashMap<>();
+    private Optional<SensiboSkyConfiguration> config = Optional.empty();
+
+    public SensiboSkyHandler(final Thing thing) {
+        super(thing);
+    }
+
+    private static String beautify(final String camelCaseWording) {
+        final StringBuilder b = new StringBuilder();
+        for (final String s : StringUtils.splitByCharacterTypeCamelCase(camelCaseWording)) {
+            b.append(" ");
+            b.append(s);
+        }
+        final StringBuilder bs = new StringBuilder();
+        for (final String t : StringUtils.splitByWholeSeparator(b.toString(), " _")) {
+            bs.append(" ");
+            bs.append(t);
+        }
+
+        return WordUtils.capitalizeFully(bs.toString()).trim();
+    }
+
+    private String getMacAddress() {
+        if (config.isPresent()) {
+            return config.get().macAddress;
+        }
+        throw new IllegalArgumentException("No configuration present");
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        handleCommand(channelUID, command, getSensiboModel());
+    }
+
+    public void updateAcState(SensiboSky sensiboSky, String property, Object value) {
+        StateChange stateChange = checkStateChangeValid(sensiboSky, property, value);
+        if (stateChange.valid) {
+            getAccountHandler().ifPresent(
+                    handler -> handler.updateSensiboSkyAcState(getMacAddress(), property, stateChange.value, this));
+        } else {
+            logger.info("Update command not sent; invalid state change for SensiboSky AC state: {}",
+                    stateChange.validationMessage);
+        }
+    }
+
+    private void updateTimer(SensiboSky sensiboSky, @Nullable Integer secondsFromNowUntilSwitchOff) {
+        getAccountHandler()
+                .ifPresent(handler -> handler.updateSensiboSkyTimer(getMacAddress(), secondsFromNowUntilSwitchOff));
+    }
+
+    @Override
+    protected void handleCommand(final ChannelUID channelUID, final Command command, final SensiboModel model) {
+        model.findSensiboSkyByMacAddress(getMacAddress()).ifPresent(sensiboSky -> {
+            if (sensiboSky.isAlive()) {
+
+                updateStatus(ThingStatus.ONLINE); // In case it has been offline
+                if (CHANNEL_CURRENT_HUMIDITY.equals(channelUID.getId())) {
```

Comment:
This would be a good place to use a switch statement instead of a if-else chain.

Reply:
Agreed